### PR TITLE
Optimize get_dataframe_data function

### PR DESCRIPTION
**Pull request by Hardcode84**
* Remove lazy unboxing
* Simplify and force inline `get_dataframe_data` function to remove excessive increfs/decrefs

Based on #663

### CODE REVIEW

File: sdc/hiframes/boxing.py [hunk 1]
```diff
@@ -100,9 +100,19 @@ def unbox_dataframe(typ, val, c):
...
+    for col_ind in range(n_cols):
+        series_obj = c.pyapi.object_getattr_string(val, typ.columns[col_ind])
+        arr_obj = c.pyapi.object_getattr_string(series_obj, "values")
+        ty_series = typ.data[col_ind]
+        if isinstance(ty_series, types.Array):
+            native_val = unbox_array(typ.data[col_ind], arr_obj, c)
+        elif ty_series == string_array_type:
+            native_val = unbox_str_series(string_array_type, series_obj, c)
+
+        dataframe.data = c.builder.insert_value(
```
Comment (AlexanderKalistratov) [inline sdc/hiframes/boxing.py]: I believe we can set data once, not in every iteration

Comment (Hardcode84) [reply sdc/hiframes/boxing.py]: This code works like this:
```
data1 = {undef,undef,undef,undef}
data2 = insert(data1, val1, 0) # {val1, undef, undef, undef}
data3 = insert(data2, val2, 1) # {val1, val2, undef, undef}
data4 = insert(data3, val3, 2) # {val1, val2, val3, undef}
...etc...
```

Comment (AlexanderKalistratov) [reply sdc/hiframes/boxing.py]: Can't it be something like:
```python
    data_tup = c.context.make_tuple(
        c.builder, types.Tuple(types), values)
```
?

Not sure how to get types though

Comment (Hardcode84) [reply sdc/hiframes/boxing.py]: meh, not worth it

**Comment by AlexanderKalistratov** [comment]
Please remove `unboxed` from DataFrame model:
https://github.com/IntelPython/sdc/blob/master/sdc/hiframes/pd_dataframe_type.py#L96

**Pull request merged**